src/components/GameView/GameMenu.vue

```diff
@@ -132,6 +143,11 @@ export default {
       this.loading = false;
       this.showStalemateDialog = false;
     },
+    async stopSpectate() {
+      this.loading = true;
+      await this.$store.dispatch('requestSpectateLeave');
+      this.$router.push('/');
```

Comment:
We should handle the possible error from the store dispatch and set the loading state to false before navigating away. In this case we want to navigate home even if there's an error so I think we can use `finally` without a catch

```suggestion
      try {
        this.loading = true;
        await this.$store.dispatch('requestSpectateLeave');
      } finally {
        this.loading = false;
        this.$router.push('/');
      }
```

Reply:
Also the loading state is currently only used in the dialog, but we don't show the dialog during this flow, so we should add a loading state into the menu itself (see my suggestion above)